### ISSUE TITLE
Parallax image results in choppy scrolling in Chrome/Safari

### ISSUE THREAD

**Issue by skawaii** [creativetimofficial/ct-material-kit-pro#28]
When using a parallax image and scrolling with a mouse wheel or by using the keyboard arrow keys, there's a choppy effect as the image is transformed. Is there anything we can do to smooth this out? It scrolls perfectly in Firefox.

Chrome 59.0.3071.115 64bit (hardware acceleration is turned on)
Mac OS X 10.12.5

Link to video illustrating the issue - https://drive.google.com/file/d/0B0-GC9_5NhWAdDUxUkt2ekRnaDA/view?usp=sharing

**Comment by alexandru-paduraru** [comment]
@skawaii thank you for the information. I've checked and looked for different libraries but couldn't find something that would work fine on keyboard click. For example this one: http://pixelcog.github.io/parallax.js/ or this one: http://www.jqueryrain.com/?ZSY7piT2 

I don't have any ideas at this moment on how we can differentiate the scroll wheel from the regular touchpad from Macbook :-( so we could write some specific code just for that case.

Any ideas on your side? Or did you see any scroll library that is working fine on your case?

Best,
Alex

**Comment by skawaii** [comment]
I just spent all morning looking around the Web to try and figure this out, but didn't really come up with much.

I looked at the two libraries you mentioned above and while I'm not noticing as much jerking motion while scrolling on their example pages, it's definitely not as nice as in Firefox. Even Google's own demo page doesn't scroll very nicely (http://googlechrome.github.io/ui-element-samples/parallax/ linked from this article - https://developers.google.com/web/updates/2016/12/performant-parallaxing). Seriously, scroll with a mouse wheel or keyboard on any of the demo pages listed in this thread in Firefox and Chrome; the difference is night and day!

Since the behavior seems to happen on pretty much every parallax site I can find, I suspect the issue is with Chrome itself. I filed a bug at https://bugs.chromium.org/p/chromium/issues/detail?id=751175, if you want to follow along.

I also found this page (http://www.abileweb.com/easy-fix-for-parallax-scrolling-backgrounds-issue-with-chrome/), but I didn't notice any difference on my site when I tried it out. Maybe I'm doing something wrong though. If you guys feel like taking a look, I'd certainly appreciate it.

BTW, my site is at https://brew-alpha.cloud.n-ask.com/ if you want to see the issue live. Maybe there's something I've overlooked to get better performance.

**Comment by alexandru-paduraru** [comment]
@skawaii got it. I was thinking the same, that can be an issue of the browser. I will check the issue that you open on Chrome community to see how that will evolve. On your example website, the page is going very very smooth on all browsers on Mac with mouse and touch bar. With the keyboard, it is having that laggy behavior. We will see what is happening on that thread :D

**Comment by skawaii** [comment]
@alexandru-paduraru Had a good chat with the chromium folks on my linked ticket. Turns out that on Windows the jerky scrolling can be avoided via a Windows or Chrome setting. While that's nice to know, it doesn't really help me in this situation (it's not like I can tell my users to change a setting when they go to my project's page). The Chrome smooth scrolling setting isn't even an option on macOS.

At this point, I'm not sure what else to do except turn off the parallax effect, at least for Chrome.

Above you said that the page is scrolling fine with all browswers on a Mac using a mouse. You're not seeing the jumpy/jerky behavior in Chrome when using a mouse wheel to scroll? Because I certainly am. In fact, that what I was doing in the video I linked above. It would be curious if we're seeing different behavior.
